src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1316,16 +1333,22 @@ private bool AreFieldsAuthorizedForEntity(string clientRole, string entityName,
                 SyntaxKind underlyingFieldKind = fieldDetails.Item2;
 
                 // For a column field, we do not have to recurse to process fields in the value - which is required for relationship fields.
-                if (GraphQLUtils.IsScalarField(underlyingFieldKind))
+                if (GraphQLUtils.IsScalarField(underlyingFieldKind) || underlyingFieldKind is SyntaxKind.NullValue)
                 {
-                    // It might be the case that we are currently processing the fields for a linking input object.
-                    // Linking input objects enable users to provide input for fields belonging to the target entity and the linking entity.
-                    // Hence the backing column for fields belonging to the linking entity will not be present in the source definition of this target entity.
+                    // This code block can be hit in 3 cases:
+                    // Case 1. We are processing a column which belongs to this entity,
+                    //
+                    // Case 2. We are processing the fields for a linking input object. Linking input objects enable users to provide
+                    // input for fields belonging to the target entity and the linking entity. Hence the backing column for fields
+                    // belonging to the linking entity will not be present in the source definition of this target entity.
                     // We need to skip such fields belonging to linking table as we do not perform authorization checks on them.
+                    //
+                    // Case 3. When a relationship field is assigned a null value. Such a field also needs to be ignored.
                     if (metadataProvider.TryGetBackingColumn(entityName, field.Name.Value, out string? _))
                     {
-                        // Only add those fields to this entity's set of fields which belong to this entity and not the linking entity.
-                        entityTofields[entityName].Add(field.Name.Value);
+                        // Only add those fields to this entity's set of fields which belong to this entity and not the linking entity,
+                        // i.e. for Case 1.
+                        entityToExposedColumns[entityName].Add(field.Name.Value);
```

Comment:
tryadd?

Reply:
This is a hashset, so we don't have a TryAdd, and we don't need one. Can safely use Add.